models/fiatUnit.ts

```
@@ -13,176 +13,142 @@ export const FiatUnitSource = {
   BNR: async (): Promise<number> => {
     try {
-      const response = await fetch('https://www.bnr.ro/nbrfxrates.xml');
-      const xmlData = await response.text();
-
       // Fetching USD to RON rate
-      const pattern = /<Rate currency="USD">([\d.]+)<\/Rate>/;
-      const matches = xmlData.match(pattern);
-
+      const xmlData = await (await fetch('https://www.bnr.ro/nbrfxrates.xml')).text();
+      const matches = xmlData.match(/<Rate currency="USD">([\d.]+)<\/Rate>/);
       if (matches && matches[1]) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)